analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd

```diff
@@ -0,0 +1,533 @@
+
+### H3 K28 mutant
+
+> * These tumors contain H3F3A K28M or HIST1H3B K28M mutations
+> * Co-occurring lesions include: ACVR1, TP53, ATRX mutations; PDGFRA amplification; PTEN loss
+> * Mutually-exclusive lesions: FGFR1 mutations/fusions (thalamic); IDH1 mutations; BRAF V600E (low-grade gliomas)
+> * Average age of 9 years
+> * Majority should be midline localized
+
+```{r}
+h3_k28_df <- all_data_df %>% 
+  # Only rows with H3 K28M mutations
+  filter(H3F3A.K28M == "Yes" | HIST1H3B.K28M == "Yes" | 
+           HIST1H3C.K28M == "Yes" | HIST2H3C.K28M == "Yes") %>%
+  # Only the relevant columns
+  select(Kids_First_Participant_ID, 
+         sample_id,
+         H3F3A.K28M,
+         HIST1H3B.K28M,
+         HIST1H3C.K28M,
+         HIST2H3C.K28M,
+         H3F3A.G35R,
+         H3F3A.G35V,
+         age_at_diagnosis_yr,
+         glioma_brain_region,
+         relevant_coding_mutations,
+         PTEN_focal_status,
+         PDGFRA_focal_status,
+         BRAF_V600E,
+         FGFR1_fusions) %>%
+  distinct()
+
+h3_k28_df
+```
+
+Mutually exclusive with the H3 G35 mutations?
+
+```{r}
+any(h3_k28_df$H3F3A.G35R == "Yes")
+```
+
+```{r}
+any(h3_k28_df$H3F3A.G35V == "Yes")
+```
+
+Are the majority midline localized?
+
+```{r}
+table(h3_k28_df$glioma_brain_region)
+```
+
+Yes.
+
+```{r}
+h3_k28_df %>%
+  group_by(relevant_coding_mutations) %>%
+  count()
+```
+
+The coding mutation list looks like what we would expect based on the description, with the exception of some BRAF mutations.
+There are no _IDH1_ mutations.
+
+```{r}
+h3_k28_df %>%
+  group_by(PTEN_focal_status, PDGFRA_focal_status) %>%
+  count()
+```
+
+The copy number information looks a bit different than what is described, but these are not lesions that must occur.
+
+```{r}
+summary(as.integer(h3_k28_df$age_at_diagnosis_yr))
+```
+
+Age is similar to what is described.
+
+```{r}
+table(h3_k28_df$FGFR1_fusions)
+```
+
+No _FGFR1_ fusions.
+
+The histologies file that gets distributed as part of the project and contains molecular subtypes contains one row per biospecimen.
+So we'll create a table that contains the biospecimen IDs and the calls.
+
+```{r}
+h3_k28_df_biospecimen <- all_data_df %>%
+  filter(H3F3A.K28M == "Yes" | HIST1H3B.K28M == "Yes" |
+           HIST1H3C.K28M == "Yes" | HIST2H3C.K28M == "Yes") %>%
+  select(Kids_First_Participant_ID,
+         sample_id,
+         Kids_First_Biospecimen_ID_DNA,
+         Kids_First_Biospecimen_ID_RNA, 
+         BRAF_V600E) %>%
+  mutate(molecular_subtype = case_when(
+    BRAF_V600E == "p.V600E" ~ "DMG, H3 K28, BRAF V600E",
+    TRUE ~ "DMG, H3 K28"
+  )) %>%
+  select(-BRAF_V600E)
+```
+
+### H3 G35 mutant
+
+> * These tumors contain H3F3A G35R/V mutations
+> * Co-occurring lesions include: ATRX/DAXX, TP53, SETD2 mutations, NTRK fusions
+> * Mutually exclusive lesions: IDH1 mutations
+> * Average age of 20 years
```

Comment:
You tested the age estimate in the previous section. Did you want to do that here as well? I don't know that it is completely necessary.

Reply:
The table in the rendered notebook has few enough rows that it's easy to tell that the ages agree with what was described.